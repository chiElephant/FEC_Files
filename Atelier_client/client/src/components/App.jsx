import React, { useState, useEffect } from 'react';
import ReviewsModule from './reviews_module/ReviewsModule.jsx'

function App(props) {
  return (
    <div>
      <h1> The RedBean Atelier App </h1>
      <ReviewsModule
        productId={71698}
        productName={'Bright Future Sunglasses'}/>
    </div>
  );
}

export default App;
