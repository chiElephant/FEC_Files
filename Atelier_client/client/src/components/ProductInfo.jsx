import React, { useState, useEffect } from 'react';
import { list_product, product_information, product_styles, related_products, product_reviews } from '../../example_data/example.js';


function ProductInfo (props) {

  console.log('These are the product features: ', props.product)

  if (Object.keys(props.product).length) {
    return (
      <div>
        <div className='product_info_reviews'>
          <div>This product has 4 stars reviews</div>
          <button>Read all reviews</button>
        </div>
        <div>{props.product.category}</div>
        <div>{props.product.name}</div>
        <div>{props.product.default_price}</div>
        <h4>{props.product.slogan}</h4>
        <div>{props.product.description}</div>
        <ul>
          {
            props.product.features.map((feature) => {
              return (
                <li>{feature.value + ' ' + feature.feature}</li>
              )
            })
          }
        </ul>


        <button>Add to My Outfit</button>

      </div>
    )
  }
}

export default ProductInfo;