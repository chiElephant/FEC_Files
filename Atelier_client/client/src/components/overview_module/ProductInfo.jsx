import React, { useState, useEffect } from 'react';
import axios from 'axios';
import getAvgRating from './helper-functions/helper.js';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { solid, regular } from '@fortawesome/fontawesome-svg-core/import.macro';
import { Link } from 'react-scroll';

function ProductInfo(props) {

  const [avgRating, setAvgRating] = useState(0);
  const [totalReviews, setTotalReviews] = useState(0);

  // Make a request to get the reviews information, calculate avg, and set state

  useEffect(() => {
    axios.get('/reviews/meta', { params: {product_id: props.product.id }})
      .then((data) => {
        // console.log('The reviews meta data is: ', data.data.ratings);
        // console.log(`The avg rating is: ${getAvgRating(data.data.ratings)}`)
        setAvgRating(getAvgRating(data.data.ratings).avg);
        setTotalReviews(getAvgRating(data.data.ratings).reviewsCount);
      })
  }, []);


  if (Object.keys(props.product).length && avgRating !== null) {
    return (
      <div>
        <div className="product_info_reviews">
          <fieldset>
            {avgRating >= 1
                ? <FontAwesomeIcon
                    id="star-1"
                    icon={solid('star')}
                    key='1'
                  />
                : <FontAwesomeIcon
                id="star-1"
                icon={regular('star')}
                key='1'
                />
            }
            {avgRating >= 2
              ? <FontAwesomeIcon
                  id="star-2"
                  icon={solid('star')}
                  key='2'
                />
              : <FontAwesomeIcon
              id="star-2"
              icon={regular('star')}
              key='2'
              />
            }
            {avgRating >= 3
              ? <FontAwesomeIcon
                  id="star-3"
                  icon={solid('star')}
                  key='3'
                />
              : <FontAwesomeIcon
              id="star-3"
              icon={regular('star')}
              key='3'
              />
            }
            {avgRating >= 4
              ? <FontAwesomeIcon
                  id="star-4"
                  icon={solid('star')}
                  key='4'
                />
              : <FontAwesomeIcon
              id="star-4"
              icon={regular('star')}
              key='4'
              />
            }
            {avgRating === 5
              ? <FontAwesomeIcon
                  id="star-5"
                  icon={solid('star')}
                  key='5'
                />
              : <FontAwesomeIcon
              id="star-5"
              icon={regular('star')}
              key='5'
              />
            }
          </fieldset>
          <Link to='reviews-module' smooth={true} duration={500}>Read all &#40;{totalReviews}&#41; reviews </Link>
        </div>
        <div>{props.product.category}</div>
        <div>{props.product.name}</div>
        <div>{props.product.default_price}</div>
        <h4>{props.product.slogan}</h4>
        <div>{props.product.description}</div>
        <ul>
          {props.product.features.map((feature, index) => {
            return <li key={index}>{feature.value + ' ' + feature.feature}</li>;
          })}
        </ul>

        <button>Add to My Outfit</button>
      </div>
    );
  } else {
    return (
      <div>
        <div className="product_info_reviews">
          <div data-testid='star-rating'>This product has 4 stars reviews</div>
          <button>Read all reviews</button>
        </div>
        <button>Add to My Outfit</button>
      </div>
    );
  }
}

export default ProductInfo;
