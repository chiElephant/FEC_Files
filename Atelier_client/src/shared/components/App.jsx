import React from 'react';
import { Route, Routes } from 'react-router-dom';
import Topbar from './Topbar.jsx';
import routes from '../routes';

export default function App({ serverData = null }) {
  return (
    <>
      <Topbar />
      <Routes>
        {routes.map((route) => {
          const { path, component: Component } = route;
          return (
            <Route
              key={path}
              path={path}
              element={<Component data={serverData} />}
            />
          );
        })}
      </Routes>
    </>
  );
}
