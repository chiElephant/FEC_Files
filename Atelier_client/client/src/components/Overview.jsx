import React, { useState, useEffect } from 'react';
import { list_product, product_information, product_styles, related_products, product_reviews } from '../../example_data/example.js';
import ProductInfo from './ProductInfo.jsx';
import Styles from './Styles.jsx';
import axios from 'axios';


function Overview (props) {

  const [product, setProduct] = useState({});
  const [styles, setStyles] = useState({});

  useEffect(() => {
    axios.get(`/products/${props.product_id}`)
      .then((data) => {
        console.log(data.data);
        setProduct(data.data);
      });
  }, []);

  useEffect(() => {
    axios.get(`/products/${props.product_id}/styles`)
      .then((data) => {
        console.log(data.data);
        setStyles(data.data);
      });
  }, []);

  if (Object.keys(product).length) {
    return (
      <div>
        <ProductInfo product={product} features={product.features}/>
        <Styles product={product} styles={styles.results}/>
      </div>
    )

  }
}

export default Overview;