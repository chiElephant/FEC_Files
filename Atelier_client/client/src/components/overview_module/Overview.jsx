/* eslint-disable react/destructuring-assignment */
import React, { useState, useEffect } from 'react';
import axios from 'axios';
import ProductInfo from './ProductInfo.jsx';
import Styles from './Styles.jsx';

// 71697
function Overview(props) {
  const [product, setProduct] = useState({});
  const [styles, setStyles] = useState({
    product_id: '',
    results: [
      {
        style_id: 1,
        name: '',
        original_price: '',
        sale_price: 0,
        default: true,
        photos: [
          {
            thumbnail_url: '',
            url: '',
          },
        ],
        skus: {},
      },
    ],
  });
  const [selectedStyle, setSelectedStyle] = useState({
    style_id: 1,
    name: '',
    original_price: '',
    sale_price: 0,
    default: true,
    photos: [
      {
        thumbnail_url: '',
        url: '',
      },
    ],
    skus: {},
  });

  useEffect(() => {
    axios.get(`/products/${props.product_id}`).then((data) => {
      // console.log(data.data);
      setProduct(data.data);
    });
    axios.get(`/products/${props.product_id}/styles`).then((data) => {
      // console.log(data.data);
      setStyles(data.data);
    });
    axios.get(`/products/${props.product_id}/styles`).then((data) => {
      console.log(data.data);
      setStyles(data.data);
    });
  }, []);

  useEffect(() => {
    let def = {};
    for (let i = 0; i < styles.results.length; i++) {
      if (styles.results[i]['default?']) {
        def = styles.results[i];
        styles.results.splice(i, 1);
        styles.results.unshift(def);
      }
    }
    setSelectedStyle(def);
  }, [styles]);

  const changeStyleSelected = (style) => {
    console.log(`The selected style is: ${style}`);
    setSelectedStyle(style);
  };

  if (Object.keys(product).length) {
    return (
      <div data-testid="overview">
        <ProductInfo
          product_id={props.product_id}
          product={product}
          features={product.features}
          style={selectedStyle}
          styles={styles}
        />
        <Styles
          product={product}
          styles={styles.results}
          changeStyleSelected={changeStyleSelected}
          style={selectedStyle}
        />
      </div>
    );
  }
}

export default Overview;
