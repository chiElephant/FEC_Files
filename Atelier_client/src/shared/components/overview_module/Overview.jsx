import React, { useState, useEffect } from 'react';
import ProductInfo from './ProductInfo.jsx';
import Styles from './Styles.jsx';
import Gallery from './Gallery.jsx';
import Slogan from './Slogan.jsx';

export default function Overview({ productData }) {
  const [selectedStyle, setSelectedStyle] = useState(productData.styles[0]);

  const changeStyleSelected = (style) => {
    setSelectedStyle(style);
  };

  return (
    <div className="main-container">
      <div data-testid="overview" className="overview">
        <Gallery selectedStyle={selectedStyle} />
        <div className="new-right">
          <ProductInfo {...productData} selectedStyle={selectedStyle} />
          <Styles
            styles={productData.styles}
            changeStyleSelected={changeStyleSelected}
            selectedStyle={selectedStyle}
          />
        </div>
      </div>
      <Slogan {...productData} />
    </div>
  );
}
