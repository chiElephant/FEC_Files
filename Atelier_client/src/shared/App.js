/* global Image */

import React, { useState, useEffect, Suspense } from 'react';
import { useParams } from 'react-router-dom';

import axios from 'axios';
import Loading from './components/Loading.jsx';
import QandAModule from './components/questions_answers_module/QandAModule.jsx';
import ReviewsModule from './components/reviews_module/ReviewsModule.jsx';
import RelatedProducts from './components/RelatedProducts.jsx';
import Overview from './components/overview_module/Overview.jsx';

function App() {
  const { id: paramId } = useParams();
  const [productData, setProductData] = useState(null);
  const [relatedProducts, setRelatedProducts] = useState({});
  // const [loading, setLoading] = useState(true);
  // const [error, setError] = useState(null);

  console.log('productData: ', productData);
  console.log('paramId: ', paramId);

  // if (loading) {
  //   return <Loading />;
  // }

  // if (error) {
  //   return <div>Error: {error.message}</div>;
  // }

  return (
    <div id="components">
      <Overview productData={productData} />
      <RelatedProducts relatedProducts={relatedProducts} />
      <QandAModule
        product_id={productData.id}
        product_name={productData.name}
        questions_answers={productData.questionsWithAnswers}
      />
      <ReviewsModule
        product_id={productData.id}
        product_name={productData.name}
        reviewsData={productData.reviews}
        metaData={productData.metaData}
      />
    </div>
  );
}

export default App;
