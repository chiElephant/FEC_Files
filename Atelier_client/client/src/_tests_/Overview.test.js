import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Overview from '../components/Overview.jsx';
import React from 'react';
import ProductInfo from '../components/ProductInfo.jsx';
import App from '../components/App.jsx';
import '@testing-library/jest-dom';

// it('Overview Component renders correctly', () => {
//   const tree = renderer.create(<Overview />).toJSON();
//   expect(tree).toMatchSnapshot();
// });


describe('Atelier topbar', () => {
    test('The Atelier Topbar renders correctly', () => {
      render (<App />);
      const element = screen.getByRole('heading', {
        name: /The RedBean Atelier App/i
      });
      expect(element).toBeInTheDocument();
  });
})


