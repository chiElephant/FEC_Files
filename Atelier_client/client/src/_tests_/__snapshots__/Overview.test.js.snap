// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview Component renders correctly 1`] = `null`;

exports[`renders correctly 1`] = `
<div>
  <div>
    <div
      className="product_info_reviews"
    >
      <div>
        This product has 4 stars reviews
      </div>
      <button>
        Read all reviews
      </button>
    </div>
    <div>
      Jackets
    </div>
    <div>
      Camo Onesie
    </div>
    <div>
      140
    </div>
    <h4>
      Blend in to your crowd
    </h4>
    <div>
      The So Fatigues will wake you up and fit you in. This high energy camo will have you blending in to even the wildest surroundings.
    </div>
    <ul>
      <li>
        Rubber Sole
      </li>
      <li>
        FullControlSkin Material
      </li>
    </ul>
    <button>
      Add to My Outfit
    </button>
  </div>
  <div
    className="product_overview_style_selector"
  >
    <div>
      <div>
        This is the style number 
        1
         
      </div>
      <img
        src="https://i1.adis.ws/i/boohooamplience/mzz12456_grey_xl_3?$product_image_main_mobile$&fmt=webp"
      />
    </div>
    <div>
      <div>
        This is the style number 
        1
         
      </div>
      <img
        src="https://i1.adis.ws/i/boohooamplience/mzz12456_grey_xl_3?$product_image_main_mobile$&fmt=webp"
      />
    </div>
    <div>
      <div>
        This is the style number 
        1
         
      </div>
      <img
        src="https://i1.adis.ws/i/boohooamplience/mzz12456_grey_xl_3?$product_image_main_mobile$&fmt=webp"
      />
    </div>
  </div>
</div>
`;
